common/useFissionAuth.ts

```diff
@@ -0,0 +1,171 @@
+import { useEffect, useState } from 'react'
+
+import * as webnative from 'webnative'
+import * as webnativeFilecoin from 'webnative-filecoin';
+
+import * as C from '@common/constants';
+import * as R from '@common/requests';
+
+import Cookies from 'js-cookie';
+
+webnative.setup.debug({ enabled: true })
+
+export function useFissionAuth() {
+  const [state, setState] = useState<webnative.State>(null)
+  let fs;
+  let authScenario: webnative.Scenario | null = null;
+  let username: string = null;
+
+
+  /** Webnative Initialization
+   * Load webnative and configure permissions.
+   */
+
+  useEffect(() => {
+    async function getState() {
+      const result = await webnative.initialise({
+        permissions: {
+          // The Estuary token is stored in app storage
+          // at bgins/estuary-www.
+          app: {
+            name: 'estuary-www',
+            creator: 'bgins',
+          },
+          fs: {
+            // The cosigner key is stored in the private filesystem
+            // at the path Keychain/fil-cosigner.
+            private: [webnativeFilecoin.DEFAULT_KEY_PERMISSION]
+          }
+        },
+      })
+      setState(result)
+    }
+
+    getState()
+  }, [])
+
+
+  /** User and filesystem initialization
+   * If the user is authenticated with Fission, set their authScenario,
+   * filesytem, and username.
+   */
+
+  switch (state?.scenario) {
+    case webnative.Scenario.AuthSucceeded:
+    case webnative.Scenario.Continuation:
+      authScenario = state.scenario;
+      fs = state.fs;
+      username = state.username;
```

Comment:
Just double checking that usernames are unique and wondering if there was a better identifier? Should we keep this on Estuary's side?

Reply:
Usernames are unique, but some of our team are doing work with distributed identities (DIDs). There may eventually be some interesting use cases for holding onto that.

For now, the username is just a convenient, human-readable thing to show users when they are coming back to Estuary from the Fission Auth Lobby. We could totally show that inside Estuary itself, and maybe we could suggest it to users for their Estuary username?

I'm not sure if there is a benefit to Estuary storing this username at the moment, but it's something we should think about down the road if Fission becomes the primary auth mechanism in Estuary.